src/app/core/submission/vocabularies/models/vocabulary-options.model.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Representing vocabulary properties
+ */
+export class VocabularyOptions {
```

Comment:
I'm not sure this class is needed.  Both the `metadata` and the `collection` params were removed from the `/submission/vocabularies/<:name>/entries` endpoint.  See the REST Contract changes here: https://github.com/DSpace/Rest7Contract/pull/128#issuecomment-656260116

This is an odd class anyways, as there's no such thing as a "VocabularyOptions" object -- there are just Vocabularies and VocabularyEntries.  But, it seems like you should be able to remove it.

Reply:
I've removed metadata and collection from requests where are not needed, they are only used for [this method](https://github.com/4Science/dspace-angular/blob/34da1dc0550d1fe80933c9124e515b7919282ff9/src/app/core/submission/vocabularies/vocabulary.service.ts#L266)

Reply:
@atarix83 : I'm confused by your answer.  That method (`searchVocabularyByMetadataAndCollection`) does not appear to be used anywhere in your code (it's only called from `vocabulary.service.spec.ts`).  That's why I feel this entire `VocabularyOptions` object is not needed...it seems to relate to a lot of code that is unused & could be removed from this PR.

Reply:
Hi @tdonohue !
The VocabularyOption class is used in 34 files, of which only 10 test files. It is true that the "collection" and "metadata" parameters have been removed from the backend but other parameters of this object are used to valorize the options in many objects and methods, and I believe that the removal of this class requires a very important refactoring without much benefit.

Reply:
@ddinuzzo : Thank you for your response.  Just a quick follow-up question.  Even if we need the `VocabularyOption` class, are you certain we still need the `metadata` and `scope` properties in this class?  I'm only finding that they are used in this [unused method](https://github.com/4Science/dspace-angular/blob/34da1dc0550d1fe80933c9124e515b7919282ff9/src/app/core/submission/vocabularies/vocabulary.service.ts#L266).   It seems a bit odd to me to have these properties as part of the `VocabularyOption` class if they are never actually used (but please correct me if I'm overlooking their purpose and they are used somewhere).

So, I'm wondering if we'd want to minimally cleanup the `VocabularyOption` class to remove the `collection` and `scope`, and cleanup the unused method & spec?   After that code cleanup, I'm OK with keeping the `VocabularyOption` class around in this PR.  I'm just hesitant to add a bunch of unused code in this PR (as unused code is not going to be 
 well maintained & can cause issues later on).